relay-server/src/endpoints/common.rs

```diff
@@ -446,6 +446,8 @@ where
                         Some(envelope) => envelope,
                         None => return Err(BadStoreRequest::RateLimited(checked.rate_limits)),
                     };
+                    // TODO: Ensure that outcomes are emitted correctly for rate-limited attachments
+                    // so that it is safe to update envelope_summary here.
```

Comment:
Technically, we will have to update the envelope summary here. We chose not to do this just yet, until the rate limiting code inside `CheckEnvelope` can emit its own outcomes.